### DISCUSSION TITLE
Support State Store with Parallel consumer

### DISCUSSION THREAD

**Discussion by colinkuo** [confluentinc/parallel-consumer#596] [Ideas] [answered]
The idea is to provide options for migrating stateful services to Parallel Consumer. The reference of the potential design could be using RocksDB to persist the data on the local disk as well as using Kafka changelog topic for disaster recovery like Kafka Streams. It would be great if windowed aggregation is supported
https://docs.confluent.io/platform/current/streams/architecture.html#state

**Comment by rkolesnev** [comment, marked as answer]
It is an interesting proposition, but honestly i think it would be easier to do it the other way around and figure out a way to add processing parallelisation to Kafka Streams instead.
